dask/array/tests/test_wrap.py

```diff
@@ -12,7 +12,7 @@ def test_ones():
     x = np.array(a)
     assert (x == np.ones((10, 10), "i4")).all()
 
-    assert a.name.startswith("ones-")
+    assert a.name.startswith("ones_like-")
```

Comment:
It would be slightly nicer on users to keep `ones` and drop the `_like` here, which I'm guessing most novice users won't understand.  

I'm looking through the code now, and I'm wondering if we can handle this in the `broadcast_trick` function by stripping off the `_like` text when we assign `inner.__name__`

Reply:
> I'm looking through the code now, and I'm wondering if we can handle this in the `broadcast_trick` function by stripping off the `_like` text when we assign `inner.__name__`

Ah cool, I didn't know how that was generated. Is there ever a case where we *wouldn't* want to strip `_like` from names?

Reply:
Most names come from calling `dask.utils.funcname` on some function that is passed in through the machinery.  There are often also ways to override that name.

> Is there ever a case where we wouldn't want to strip _like from names?

Maybe if the user explicitly called `da.ones_like` themselves?  But even in that case I can't imagine that they would care.  For reference, these names end up showing up in UI elements like progress bars and the dashboard.

Reply:
> > Is there ever a case where we wouldn't want to strip _like from names?
> 
> Maybe if the user explicitly called `da.ones_like` themselves? But even in that case I can't imagine that they would care. For reference, these names end up showing up in UI elements like progress bars and the dashboard.

For reference, currently when a user calls `da.ones_like` then they see `ones-` instead of `ones_like-` in the dask task names. Stripping the `_like` out will retain this current behaviour.